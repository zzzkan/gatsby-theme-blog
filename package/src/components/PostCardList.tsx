import React from "react"
import { Box, SimpleGrid } from "@chakra-ui/react"
import { PostCard, PostCardProps } from "../components/PostCard"

export type PostCardListProps = {
  readonly posts: ReadonlyArray<PostCardProps | null> | null
}

export const PostCardList: React.FC<PostCardListProps> = ({ posts }) => {
  if (posts == null || posts.length === 0) return null
  return (
    <SimpleGrid columns={[1, 2, 3]} spacing={"6"}>
      {posts.map((post) => {
        if (post == null) return <Box>Unknown Post</Box>
        return <PostCard key={post.slug} {...post} />
      })}
    </SimpleGrid>
  )
}
