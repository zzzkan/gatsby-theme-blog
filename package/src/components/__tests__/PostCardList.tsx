import React from "react"
import { render } from "@testing-library/react"
import { PostCardList } from "../PostCardList"

describe("PostCardList component", () => {
  vi.mock("../PostCard", () => {
    return {
      PostCard: vi.fn().mockImplementation(() => <div>PostCard</div>),
    }
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  afterAll(() => {
    vi.restoreAllMocks()
  })

  test("snapshot", () => {
    const { asFragment } = render(
      <PostCardList
        posts={[{ slug: "post1" }, { slug: "post2" }, { slug: "post3" }] as any}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should not render anything if props is null", () => {
    const { container } = render(<PostCardList posts={null} />)
    expect(container.firstChild).toBeNull()
  })

  test("should not render anything if props is []", () => {
    const { container } = render(<PostCardList posts={[]} />)
    expect(container.firstChild).toBeNull()
  })

  test("should render multiple post card", () => {
    const { getAllByText } = render(
      <PostCardList
        posts={[{ slug: "post1" }, { slug: "post2" }, { slug: "post3" }] as any}
      />
    )
    expect(getAllByText("PostCard").length).toBe(3)
  })
})
