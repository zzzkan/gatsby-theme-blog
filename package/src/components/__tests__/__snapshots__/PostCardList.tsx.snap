// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostCardList component > snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-gap: 6px;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  grid-template-columns: repeat(2, minmax(0, 1fr));
  grid-template-columns: repeat(3, minmax(0, 1fr));
}

<div
    class="emotion-0"
  >
    <div>
      PostCard
    </div>
    <div>
      PostCard
    </div>
    <div>
      PostCard
    </div>
  </div>
</DocumentFragment>
`;
